### PR TITLE
Add int8 (since gdal 3.7.0)

### PR DESCRIPTION
**Pull request by vincentvaroquauxads**

### CODE REVIEW

**Comment by vincentvaroquauxads** [comment]
Unfortunately, I did not find a way to have a compile-time error iif Int8 is used with Gdal < 3.7.0. It will result in a runtime error.

File: doc_test.go [hunk 1]
```diff
@@ -289,6 +289,19 @@ func Example_rasterTutorial() {
 	poDstDS.Bands()[0].Write(0, 0, abyRaster, 512, 512)
 	poDstDS.Close()
 
+	poDstDSInt8, err := godal.Create(godal.GTiff, pszDstFilename, 1, godal.Int8, 512, 512)
```
Comment (tbonfort) [inline doc_test.go:292]: Please put this test inside a new function in godal_test.go. doc_test.go is meant for documenting basic usage

File: godal.h [hunk 2]
```diff
@@ -25,6 +25,10 @@
 
 #if GDAL_VERSION_NUM < 3000000
 	#error "this code is only compatible with gdal version >= 3.0"
+#elif GDAL_VERSION_NUM < GDAL_COMPUTE_VERSION(3, 7, 0)
```
Comment (tbonfort) [inline godal.h:28]: I would prefer if this went into its own #if instead of this #elif, as the intent is different

File: godal_test.go [hunk 3]
```diff
@@ -223,6 +229,19 @@ func TestCreate(t *testing.T) {
...
+	defer os.Remove(tmpname3)
+	ds, err = Create(GTiff, tmpname3, 1, Int8, 20, 20, ErrLogger(ehc.ErrorHandler))
+	if CheckMinVersion(3, 7, 0) {
+		assert.NoError(t, err)
+
+		err = ds.Close(ErrLogger(ehc.ErrorHandler))
+		assert.NoError(t, err)
+	} else if err == nil {
+		panic("godal.Int8 is not supported with GDAL<3.7.0, but godal.Create returns no error")
+	}
```
Comment (tbonfort) [inline godal_test.go:244]: ``
```suggestion
	} else {
	        assert.Error(t, err, "godal.Int8 is not supported with GDAL<3.7.0, but godal.Create did not return an error")
	}
```

**Pull request merged**